core/source_generator/source_generator.cc

```diff
@@ -71,13 +71,22 @@ string prettySigForMethod(const core::GlobalState &gs, core::MethodRef method, c
     if (!flags.empty()) {
         flagString = fmt::format("{}.", fmt::join(flags, "."));
     }
+    string typeParamString = "";
+    if (!sym->typeArguments().empty()) {
+        for (const auto &typeParam : sym->typeArguments()) {
+            typeParams.push_back(fmt::format(":{}", typeParam.data(gs)->name.show(gs)));
+        }
+        typeParamString = fmt::format("type_parameters({}).", fmt::join(typeParams, ", "));
+    }
     string paramsString = "";
     if (!typeAndArgNames.empty()) {
         paramsString = fmt::format("params({}).", fmt::join(typeAndArgNames, ", "));
     }
 
-    auto oneline = fmt::format("{} {{ {}{}{} }}", sigCall, flagString, paramsString, methodReturnType);
-    if (oneline.size() <= MAX_PRETTY_WIDTH && typeAndArgNames.size() <= MAX_PRETTY_SIG_ARGS) {
+    auto oneline =
+        fmt::format("{} {{ {}{}{}{} }}", sigCall, flagString, typeParamString, paramsString, methodReturnType);
+    if ((oneline.size() - typeParamString.size()) <= MAX_PRETTY_WIDTH &&
+        typeAndArgNames.size() <= MAX_PRETTY_SIG_ARGS) {
```

Comment:
I wonder if we should refactor this to do the math before allocating the string.